src/App.tsx

```diff
@@ -1,26 +1,112 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { FC, ReactNode, useMemo, useState } from 'react';
+
+import { useGetReposByLoginQuery, useGetUserByLoginQuery } from 'api/user.api';
+import DefaultWindow from 'components/layout/DefaultWindow';
+import Header from 'components/layout/Header';
+import Profile from 'components/layout/Profile';
+import Repositories from 'components/layout/Repositories';
+import Container from 'components/shared/Container';
+
+import 'styles/components/app.scss';
+
+const App: FC = () => {
+  const [searchProfile, setSearchProfile] = useState('');
+  const [currentPage, setCurrentPage] = useState(1);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(undefined);
+
+  const user = useGetUserByLoginQuery(searchProfile, {
+    skip: !searchProfile,
+  });
+
+  const repos = useGetReposByLoginQuery(
+    {
+      login: searchProfile,
+      page: currentPage,
+    },
+    { skip: !searchProfile },
+  );
+
+  const {
+    currentData: userData,
+    isFetching: isFetchingUser,
+    error: errorUser,
+    isError: isErrorUser,
+  } = user;
+
+  const {
+    currentData: reposData,
+    isFetching: isFetchingRepos,
+    error: errorRepos,
+    isError: isErrorRepos,
+  } = repos;
+
+  const changePage = (page: number): void => {
+    return setCurrentPage(page);
+  };
+
+  const getRepositories = (): ReactNode | null => {
+    const repositories = reposData ?? null;
+    const totalRepos = userData?.public_repos ?? 0;
+
+    if (repositories) {
+      return (
+        <Repositories
+          changePage={changePage}
+          repositories={repositories}
+          currentPage={currentPage}
+          totalRepos={totalRepos}
+        />
+      );
+    }
+
+    return null;
+  };
+
+  const getContent = (): ReactNode | null => {
+    if (!userData) {
+      return <DefaultWindow loading={isLoading} error={error} />;
+    }
+
+    return (
+      <div className="app__content">
+        <Container>
+          <div className="app__info-container">
+            {userData && <Profile user={userData} />}
+
+            {getRepositories()}
+          </div>
+        </Container>
+      </div>
+    );
+  };
+  const handleSearchInputChange = (value: string): void => {
+    setSearchProfile(value);
+    setCurrentPage(1);
+  };
+
+  useMemo(() => {
+    if (isFetchingUser || isFetchingRepos) {
+      setIsLoading(true);
+    } else {
+      setIsLoading(false);
+    }
+
+    if (isErrorUser || isErrorRepos) {
+      // @ts-ignore
+      return isErrorUser ? setError(errorUser) : setError(errorRepos);
+    }
+
+    return null;
+  }, [isFetchingUser, isFetchingRepos, isErrorUser, isErrorRepos]);
```

Comment:
why useMemo and remove @ts-ignore

Reply:
I wanted to optimize and replaced useEffect with useMemo to achieve less rendering. But after studying it, I realized that it is not quite correct. I put @ts-ignore because I couldn't solve the problem of typing the passed error object from RTK query, I rewrote this logic a bit and tried to type it according to rtk-query documentation.